src/main/java/vct/col/newrewrite/SpecifyImplicitLabels.scala

```diff
@@ -0,0 +1,45 @@
+package vct.col.newrewrite
+
+import vct.col.ast.RewriteHelpers._
+import vct.col.ast._
+import vct.col.ast.AstBuildHelpers._
+
+import scala.collection.mutable
+
+case class SpecifyImplicitLabels() extends Rewriter {
+  val labelStack = new mutable.Stack[LabelDecl]()
+
+  override def dispatch(stat: Statement): Statement = stat match {
+    case block@Block(Seq(Label(labelDecl), _: Loop)) =>
+      labelStack.push(labelDecl)
+      val res = rewriteDefault(block)
+      labelStack.pop()
+      res
+    case block@Block(Seq(Label(labelDecl), _: Switch)) =>
+      labelStack.push(labelDecl)
+      val res = rewriteDefault(block)
+      labelStack.pop()
+      res
+    case loop: Loop =>
+      implicit val o = loop.o
+      val labelDecl = new LabelDecl()(SourceNameOrigin("loop", o))
```

Comment:
I intended to use SourceNameOrigin strictly for names that come from source code. I think it's a good idea to implement a custom origin for each pass and/or generation concept, so we can track derivations precisely (and preferably layered, so it describes itself as something like "the label for the loop at $x" or so).